cmd/completion.go

```diff
@@ -0,0 +1,267 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+
+const completionDesc = `
+Generate autocompletions script for Astro for the specified shell (bash or zsh).
+
+This command can generate shell autocompletions. e.g.
+
+	$ astro completion bash
+
+Can be sourced as such
+
+	$ source <(astro completion bash)
+`
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+)
+
+func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
```

Comment:
We should be able to use `ExactArgs(1)`  validator here to have cobra handle this for us

```go

	cmd := &cobra.Command{
		Use:   "completion SHELL",

		Short: "Generate autocompletions script for the specified shell (bash or zsh)",
		Long:  completionDesc,
		RunE: func(cmd *cobra.Command, args []string) error {
			return runCompletion(os.Stdout, cmd, args)
		},
		Args: cobra.ExactValidArgs(1),
		ValidArgs: shells,
	} 
```

Reply:
fixed.